### ISSUE TITLE
Question about obtaining ground-truth image intrinsics in the EgoExo4D Hand Pose Challenge

### ISSUE THREAD

**Issue by 1a2a3x** [geopavlakos/hamer#112]
![Image](https://github.com/user-attachments/assets/5ebe1b4b-57b9-40e6-89bd-1416207b2a0d)
Thank you for sharing your impressive work on hand pose estimation! I’m a beginner in this field and have been studying your ablation results in Table 1. I noticed that the "HaMeR-align" method uses "ground-truth image intrinsics" to optimize rotation and translation.

Could you kindly explain how these ground-truth intrinsics were obtained in the EgoExo4D dataset? For example, were they provided as part of the dataset, calibrated separately, or derived through another process? I apologize if this is a basic question—I’m still learning about these concepts and would greatly appreciate any clarification.

Thank you for your time!

**Comment by geopavlakos** [comment]
The ground-truth intrinsics were provided by EgoExo4D.

**Comment by 1a2a3x** [comment]
@geopavlakos 

> The ground-truth intrinsics were provided by EgoExo4D.
Thank you very much for your response. My understanding is that by inputting the focal length and principal points from the provided intrinsic matrix into the pred_cam_t_full function and performing rotation and translation, the resulting data does not show a significant reduction in MPJPE (compared to root node subtraction). Could you please advise on what steps I might be missing?

![Image](https://github.com/user-attachments/assets/35c6cb29-62cb-4f7b-847e-0ceec3c506a0)

**Comment by 1a2a3x** [comment]
> I'm not sure I fully understand your question, but what the table shows is that the MPJPE metric improves significantly after applying the rotation and translation optimization. The PA-MPJPE metric will not improve because it's only affected by the local hand pose which is not changing if we only optimize the rotation and translation of the hand.

Thank you very much for your response. To better illustrate my issue, I've conducted two experiments:

First, without any rotation or translation optimization (only using root-relative alignment for MPJPE calculation), I obtained an error value of 56.47mm.
`            for n in range(batch_size):
                if is_right[n] == 0:
                    pre_hand_3d = out['pred_keypoints_3d'][n].detach().cpu().numpy()
                    pre_hand_3d[:, 0] *= -1

            pre_hand_3d -= pre_hand_3d[:,0,:]
            valid_pose_3d_gt -= valid_pose_3d_gt[:,0,:]#root-relative alignment
            
            epoch_loss_3d_pos.append(
                mpjpe(pre_hand_3d * 1000, valid_pose_3d_gt * 1000).item()
            )`
Second, I incorporated the pred_cam_t_full function to optimize rotation and translation, using the focal length (150) and principal point (255.5) from the EgoExo4D dataset, and then calculated MPJPE using the following code. The final error was 55.65mm.
`pred_cam_t_full = cam_crop_to_full(pred_cam, box_center, box_size, img_size, 150.0).detach().cpu().numpy()
            for n in range(batch_size):
                if is_right[n] == 0:
                    pre_hand_3d = out['pred_keypoints_3d'][n].detach().cpu().numpy()
                    pre_hand_3d[:, 0] *= -1
                    pre_hand_3d += pred_cam_t_full[n]#optimize rotation and translation

            epoch_loss_3d_pos.append(
                mpjpe(pre_hand_3d * 1000, valid_pose_3d_gt * 1000).item()
            )`
Clearly, using pred_cam_t_full didn't effectively optimize the rotation and translation as expected. Did I miss any crucial steps in this process?

**Comment by geopavlakos** [comment]
There is no explicit translation and rotation optimization happening in this code.
`pred_cam_t_full` just updates the hand translation estimate using a heuristic, so that the hand projects well on the image under different intrinsics. This should not affect MPJPE or PA-MPJPE.

**Comment by 1a2a3x** [comment]
> There is no explicit translation and rotation optimization happening in this code. just updates the hand translation estimate using a heuristic, so that the hand projects well on the image under different intrinsics. This should not affect MPJPE or PA-MPJPE.`pred_cam_t_full`

Thank you for your explanation. Could you please advise me on how to specifically optimize the rotation and translation? Would you be able to share the code or web references you use for this?
